xbmc/addons/AddonDatabase.cpp

```diff
@@ -823,3 +830,48 @@ bool CAddonDatabase::RemovePackage(const std::string& packageFileName)
   return ExecuteQuery(sql);
 }
 
+bool CAddonDatabase::AddSystemAddon(const std::string &addonID)
+{
+  try
+  {
+    if (NULL == m_pDB.get())
+      return false;
+    if (NULL == m_pDS.get())
+      return false;
+
+    if (!IsSystemAddonKnown(addonID)) // Enabled
+    {
+      std::string sql = PrepareSQL("insert into system(id, addonID) values(NULL, '%s')", addonID.c_str());
```

Comment:
ah sorry, thought it was a loop. then you could save the extra NULL checks by using `ExecuteQuery()` still ;)